src/qaqc/01-organize_raw_data.Rmd

```diff
@@ -20,83 +20,96 @@ all_data <- list.files(path = 'data/api/', full.names = TRUE) %>%
-                             cond_pre = str_match(conductivity,paste0(str_match(conductivity,
-                                                                                "PreMeasurementActual\\s*(.*?)\\s*SpecificConductivity")[,2],"SpecificConductivity\\s*(.*?)\\s*µS/cmPost"))[,2],
-                             cond_post = str_match(conductivity,paste0(str_match(conductivity,
-                                                                                 "PostMeasurementActual\\s*(.*?)\\s*SpecificConductivity")[,2],"SpecificConductivity\\s*(.*?)\\s*µS/cm"))[,2],
-
-                             #Turbidity
-                             ntu_slope = str_match(turbidity, "Slope\\s*(.*?)\\s*Offset")[,2],
-                             ntu_offset = str_match(turbidity, "Offset\\s*(.*?)\\s*NTU")[,2],
-                             ntu_10 = str_match(turbidity, "CalibrationPoint1PreMeasurement\\s*(.*?)\\s*NTUPost")[,2],
-                             ntu_100 = str_match(turbidity, "CalibrationPoint2PreMeasurement\\s*(.*?)\\s*NTUPost")[,2],
-
-                             #Factory Defaults
-                             factory_defaults = paste0(ifelse(is.na(ntu_slope), "Turbidity ", ""),
-                                                       ifelse(is.na(rdo_slope), "RDO ", ""),
-                                                       ifelse(is.na(ph_slope_post), "pH ", ""),
-                                                       ifelse(is.na(orp_offset), "ORP ", ""),
-                                                       ifelse(is.na(cond_post), "Conductivity ", ""))) %>%
-      select(-c(ph_7_nice, ph_7_other))
-  }
-
-  cal_table <- bind_rows(cal_table) %>%
-    distinct(.keep_all = TRUE) %>%
-    group_by(site) %>%
-    mutate(DT = round_date(DT, "15 minutes")) %>%
-    padr::pad(by = 'DT', group = 'site')
+cal_files <- list.files("data/calibration_reports", pattern=".html")
+
+cal_tabler <- function(cal_files){
+  
+  cal <- rvest::read_html(paste0("data/calibration_reports/", cal_files)) %>%
+    rvest::html_nodes('div') %>%
+    rvest::html_text() %>%
+    as_tibble()
+  
+  rdo <- cal %>% filter(grepl("RDO", value)) %>% pull() %>% str_replace_all(., " ", "")
+  
+  ph_orp <- cal %>% filter(grepl("pH/ORP", value)) %>% pull() %>% str_replace_all(., " ", "")
+  
+  conductivity <- cal %>% filter(grepl("Conductivity",value)) %>% pull() %>% str_replace_all(., " ", "")
+  
+  turbidity <- cal %>% filter(grepl("Turbidity",value)) %>% pull() %>% str_replace_all(., " ", "")
+  
+  time_mst <- paste0(str_sub(cal_files, -13, -12),":", str_sub(cal_files, -11, -10))
+  
+  date <- paste0(str_sub(cal_files, -22, -19),"-", str_sub(cal_files, -18, -17),"-", str_sub(cal_files, -16, -15))
+  
+  cal_table <- tibble(site = sub("\\_.*", "", cal_files),
+                      
+                      DT = ymd_hm(paste0(date, " ", time_mst)),
```

Comment:
- consider adding use of the 'tz' argument here, as these are technically stored as 'UTC'. Use of tz will facilitate any future lubridate tz adjustments (if necessary)
- because ymd_hm() is ridiculously good at parsing, you can eliminate some characters here: DT = ymd_hm(paste(date, time_mst)) and it will parse correctly.